pkg/api/agentPoolOnlyApi/v20180331/types.go

```diff
@@ -183,6 +183,9 @@ type AgentPoolProfile struct {
 	StorageProfile string `json:"storageProfile" validate:"eq=ManagedDisks|len=0"`
 	VnetSubnetID   string `json:"vnetSubnetID,omitempty"`
 	MaxPods        *int   `json:"maxPods,omitempty"`
+	MaxCount       *int   `json:"maxCount,omitempty"`
```

Comment:
@feiskyer thoughts on this? I'm doing it different than you did for acs-engine because I wanted to support a world with multiple agent pools.

Reply:
Also in AKS I can't deploy it via the addon manager. @weinong also curious on your thoughts

Reply:
Though it is offically an addon so possibly I should move it back to addons. Thoughts? @feiskyer @weinong @amanohar

Reply:
Looks ok to me. But notice that the customers may want to change maxCount/minCount after deployment.

Reply:
Definitely. I was just going to handle that in the service as it already has merge logic that is more complete than the half hearted merge here. That is also why I didn't try to add validation.